src/main/kotlin/org/oso/core/services/impl/DefaultDeviceService.kt

```diff
@@ -25,13 +25,12 @@ class DefaultDeviceService
     }
 
     override fun createIfMissing(name: String, description: String, deviceType: DeviceType?): Device {
-        var dev = deviceRepository.findByName(name)
+        var dev = deviceRepository.findById(name).get()
```

Comment:
The abbreviation ``dev`` is not really clear in my opinion. It could have multiple meanings and therefore I would rename it to ``device``.

Reply:
👍